torch/testing/_internal/schema_check_mode.py

```diff
@@ -34,11 +35,17 @@ def display_ops(self):
         print(*self.ops, sep=",")
 
     def __torch_dispatch__(self, func, types, args=(), kwargs=None):
-        def has_mutated(before, after):
-            return not torch.equal(before, after) if isinstance(before, torch.Tensor) and isinstance(after, torch.Tensor) else False
+        def has_mutated(before, after, md):
+            if type(before) == torch.Tensor and type(after) == torch.Tensor:
+                return not (
+                    torch.equal(before, after) and
+                    md[0] == after.stride() and
+                    md[1] == after.storage()._cdata
+                )
+            return False
 
         def has_aliased(lhs, rhs):
-            return torch._C._is_alias_of(lhs, rhs) if isinstance(lhs, torch.Tensor) and isinstance(rhs, torch.Tensor) else False
+            return torch._C._is_alias_of(lhs, rhs) if type(lhs) == torch.Tensor and type(rhs) == torch.Tensor else False
```

Comment:
Why did you change this from an `isinstance()` check to instead check type equality? 

Also, this line is very long and should just be a normal if statement.

Reply:
This line was changed from isinstance() to check type equality because it is possible for a subclass of torch.Tensor to pass through the unwrapping due to not having an "elem" instance variable which will then error out on the subsequent mutation check / aliasing check. Changing the checks to type equality ensures this doesn't happen.

Reply:
Do you have some example code handy for where this would happen? I assume that we want to still check for aliasing information for tensor subclasses. 

(Also, this is good stuff to add as a comment to this function).

Reply:
It seems to happen with a "Parameter" object which has no attribute 'elem' when calling the batch norm op. For all other tensor subclasses which have a tensor within them, aliasing information is checked as expected.